RunningOrder/CustomWindow.swift

```diff
@@ -10,32 +10,123 @@ import Cocoa
 
-        return item
+        var toolbarItem = NSToolbarItem()
+
+        if itemIdentifier == .addStory {
+            toolbarItem = customToolbarButtonItem(itemIdentifier: itemIdentifier.rawValue,
+                                                  label: NSLocalizedString("Add a story", comment: ""),
+                                                  paletteLabel: NSLocalizedString("Add a story", comment: ""),
+                                                  toolTip: NSLocalizedString("Add a story", comment: ""),
+                                                  iconImageName: NSImage.addTemplateName,
+                                                  action: #selector(addStoryActionPlaceHolder))
+        }
+
+        return toolbarItem
     }
 
-    override func windowDidLoad() {
-        super.windowDidLoad()
+    func toolbarWillAddItem(_ notification: Notification) {
+        guard let userInfo = notification.userInfo else { return }
+        let item = userInfo["item"] as? NSToolbarItem
+        if let item = item, item.itemIdentifier == .toggleSidebar {
+            item.action = #selector(toggleSidebar)
+        }
+    }
 
-        window?.contentViewController = NSHostingController(rootView: MainView())
+    func validateToolbarItem(_ item: NSToolbarItem) -> Bool {
+        switch item.itemIdentifier {
+        case .addStory:
+            return false // Implement addStory button validation
+        default:
+            return true
+        }
+    }
+
+    func customToolbarButtonItem(
+        itemIdentifier: String,
+        label: String,
+        paletteLabel: String,
+        toolTip: String,
+        iconImageName: String,
+        action: Selector) -> NSToolbarItem {
+
+        let toolbarItem = CustomToolbarItem(itemIdentifier: NSToolbarItem.Identifier(rawValue: itemIdentifier))
+
+        let iconImage = NSImage(named: iconImageName)
+
+        let button = NSButton(frame: NSRect(x: 0, y: 0, width: 40, height: 40))
+        button.title = ""
+        button.image = iconImage
+        button.bezelStyle = .texturedRounded
+
+        button.action = action
+        toolbarItem.view = button
+        toolbarItem.label = label
+        toolbarItem.paletteLabel = paletteLabel
+        toolbarItem.toolTip = toolTip
+        toolbarItem.target = self
+
+        return toolbarItem
+    }
+
+// TODO change these lines in the future
+    @objc func toggleSidebar(_ sender: Any) {
+        ((window?.contentView?.subviews.first?.subviews.first?.subviews.first as? NSSplitView)?.delegate as? NSSplitViewController)?.toggleSidebar(self)
+    }
+
+    @objc func addStoryActionPlaceHolder() {
+        print("addStory toolbarbutton clicked")
+    }
+}
+
+private extension NSToolbarItem.Identifier {
+    static let addStory = NSToolbarItem.Identifier(rawValue: "AddStory")
+}
+
+class CustomToolbarItem: NSToolbarItem {
+
+    override func validate() {
+        if let control = self.view as? NSControl, let action = self.action,
```

Comment:
> For conciseness, avoid using self since Swift does not require it to access an object's properties or invoke its methods.
Use self only when required by the compiler (in @escaping closures, or in initializers to disambiguate properties from arguments). In other words, if it compiles without self then omit it.

We try to respect some conventions:

https://github.com/raywenderlich/swift-style-guide

https://swift.org/documentation/api-design-guidelines/

Reply:
😠